helix-term/src/commands.rs

```diff
@@ -1224,6 +1224,9 @@ fn replace(cx: &mut Context) {
     // need to wait for next key
     cx.on_next_key(move |cx, event| {
         let (view, doc) = current!(cx.editor);
+
+        // `encode_utf8` gives a `&mut str` which differs from the other branches.
+        #[allow(clippy::redundant_slicing)]
```

Comment:
actually, I think it would fix the issue if you change the let to `let ch: &str = match ..`